src/deck.tsx

```diff
@@ -15,11 +18,37 @@ class CardDeck extends React.Component<{}, DeckProps>{
 
 	constructor( props: any )
 	{
+
         super( props );
-        for (var i = 0; i < 52; i++) {
+        for (let i = 0; i < 52; i++) {
+            remainingCards.push(i);
+            this.shuffle();

```

Comment:
Should the shuffle be called every iteration or is it enough to call it once at the end? Like you do in "gather"

Reply:
I'll push that fix, yea that's a mistake